### PR TITLE
Implemented saving textfields input

### PR DESCRIPTION
**Pull request by 0xifarouk**

### CODE REVIEW

File: DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift [hunk 1]
```diff
@@ -145,3 +186,15 @@ class CalculatorVC: BaseViewController {
         }
     }
 }
+
+enum textFieldName: String {
```
Comment (cs4alhaider) [inline DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift]: `enum` should start with uppercase like `TextFieldName` and it's better to make it private as you don't need them anywhere else

File: DhawaAlkhreej/DhawaAlkhreej/Support/Keys/Keys+UserDefaults.swift [hunk 2]
```diff
@@ -27,5 +27,8 @@ extension Keys {
         
         /// set & get the saved|fav items - "DhawaAlkhreej.UserDefaults.savedItems"
         static let savedItems = "DhawaAlkhreej.UserDefaults.savedItems"
```
Comment (cs4alhaider) [inline DhawaAlkhreej/DhawaAlkhreej/Support/Keys/Keys+UserDefaults.swift]: Delete this key as we won't need it

File: DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift [hunk 3]
```diff
@@ -91,6 +91,43 @@ class CalculatorVC: BaseViewController {
...
+            requestedTahsilyTF.text = data[textFieldName.requestedTahsilyTF.rawValue]
+            requestedStepExamTF.text = data[textFieldName.requestedStepExamTF.rawValue]
+            accoplishedThanawiyahTF.text = data[textFieldName.accoplishedThanawiyahTF.rawValue]
+            accomplishedQuodratTF.text = data[textFieldName.accomplishedQuodratTF.rawValue]
+            accomplishedTahsilyTF.text = data[textFieldName.accomplishedTahsilyTF.rawValue]
+            accomplishedStepExamTF.text = data[textFieldName.accomplishedStepExamTF.rawValue]
+        }
+
+        DispatchQueue.main.async {
+            self.validateTextfieldInput(self.accoplishedThanawiyahTF)
```
Comment (cs4alhaider) [inline DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift:111]: Could you move this to a new method as it's better to have one job only for every method

File: DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift [hunk 4]
```diff
@@ -91,6 +91,43 @@ class CalculatorVC: BaseViewController {
         accomplishedStepExamTF.tintColor = UIColor.primary
     }
     
+    func saveTextfieldData(textFieldName: textFieldName, data: String) {
```
Comment (cs4alhaider) [inline DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift]: How about naming the method like `func saveTextFieldData(_ name: textFieldName, data: String)` ?

File: DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift [hunk 5]
```diff
@@ -50,7 +50,7 @@ class CalculatorVC: BaseViewController {
         setupCalculateButton()
         setupCalculatedPercentageLabel()
         setupTextfeilds()
-        
+        retrieveTextfieldsData()
         DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```
Comment (cs4alhaider) [inline DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift:54]: Remove this line if you don't need it

File: DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift (hunk 1 above)
Comment (0xifarouk) [reply DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift]: it's not private because I am using it in the extension.

**Pull request merged**